### PR TITLE
cancel order and show order details

### PR DESCRIPTION
**Pull request by caovanson06112001**
## Related Tickets
- [#TicketID](https://edu-redmine.sun-asterisk.vn/issues/52620)

## WHAT (optional)
- cancel order and show order details

## HOW
- cancel order and show order details

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
image
image

### CODE REVIEW

**Comment by caovanson06112001** [comment]
ready

File: app/javascript/admin/index.min.js [hunk 1]
```diff
@@ -133,6 +133,7 @@ $( document ).on("turbolinks:load", function() {
     var start = $("#date_start").val()
     var end = $("#date_end").val()
     var note = $("#note_user").val()
+    var st = "pending"
```
Comment (manhnt-0960) [inline app/javascript/admin/index.min.js:136]: ghi rõ status ra e, sao phải st

File: app/models/book.rb [hunk 2]
```diff
@@ -21,4 +21,13 @@ class Book < ApplicationRecord
   validates :quantity, presence: true, numericality: {only_integer: true}
 
   scope :latest_books, ->{order created_at: :desc}
+
+  scope :get_books, ->(id){where(category_books: {category_id: id})}
```
Comment (manhnt-0960) [inline app/models/book.rb:25]: by_ids, scope trong model book thì chắc chắn là để get book rồi, đặt tên scope như này gây khó hiểu

File: app/models/book.rb [hunk 3]
```diff
@@ -21,4 +21,13 @@ class Book < ApplicationRecord
...
+
+  scope :get_order_book, ->(id){where(order_details: {order_id: id})}
```
Comment (manhnt-0960) [inline app/models/book.rb:27]: by_order

File: config/routes.rb [hunk 4]
```diff
@@ -15,6 +15,7 @@
     scope module: "website" do
       root "home#index"
       get "/product/:id", to: "products#show", :as => :product
+      get "/category/:id", to: "products#index", :as => :category
```
Comment (manhnt-0960) [inline config/routes.rb:18]: sao k dùng resource

**Comment by caovanson06112001** [comment]
ready

**Pull request merged**